packages/typegpu/src/resolutionCtx.ts

```diff
@@ -462,38 +462,43 @@ export function resolve(
 
   const automaticIds = naturalsExcept(takenIndices);
 
+  const createCatchallGroup = () => {
+    const catchallIdx = automaticIds.next().value;
+    const catchallLayout = bindGroupLayout(Object.fromEntries(layoutEntries));
+    bindGroupLayouts[catchallIdx] = catchallLayout;
+    code = code.replaceAll(CATCHALL_BIND_GROUP_IDX_MARKER, String(catchallIdx));
+    return [
+      catchallIdx,
+      catchallLayout.populate(
+        Object.fromEntries(
+          ctx.fixedBindings.map(
+            (binding, idx) =>
+              // biome-ignore lint/suspicious/noExplicitAny: <it's fine>
+              [String(idx), binding.resource] as [string, any],
+          ),
+        ),
+      ),
+    ] as [number, TgpuBindGroup];
+  };
+
+  const layoutEntries = ctx.fixedBindings.map(
+    (binding, idx) =>
+      [String(idx), binding.layoutEntry] as [string, TgpuLayoutEntry],
+  );
```

Comment:
Could we move this definition before `createCatchallGroup`? It might be easier to reason about the dependencies between values, as `createCatchallGroup` uses `layoutEntries` from the outer scope.